jobs/haproxy/spec

```diff
@@ -39,6 +39,28 @@ consumes:
     optional: true
 
 properties:
+  ha_proxy.pre_start_script:
+    description: |
+      This script will be appended to the pre-start script and run before the job starts.
+      The pre-start script allows the job to prepare the machine before starting HAProxy,
+      for example, by setting the MTU to a custom value.
+    default: ~
+    example: |
+      # customize MTU
+      CUST_MTU=((custom_mtu))
+      INTERFACE=$(ip -4 route get 8.8.8.8 | grep -Po '(?<=dev )\S+') #'
+      CURR_MTU=$(ip link show $INTERFACE | grep -Po 'mtu \K\d+')
+      if [[ $CURR_MTU -ne $CUST_MTU ]]; then
+        sudo ip link set dev $INTERFACE mtu $CUST_MTU
+        echo "MTU changed from $CURR_MTU to $CUST_MTU, interface: $INTERFACE."
+      fi
+      CURR_MTU=$(ip link show $INTERFACE | grep -Po 'mtu \K\d+')
+      echo "MTU: $CURR_MTU, interface: $INTERFACE"
+  ha_proxy.config_mode:
```

Comment:
Can you please elaborate a bit more on your use-case requiring this additional switch? Maybe you can provide your concrete example? The logic around these options is already not trivial and I am not sure if this would be backwards compatible for all users.

In general, I'd prefer a PR per feature to be flexible for fixes when unforseen problems are caused for some stakeholders. If we can ensure backwards compatibility, we can also make an exception for those mid-sized enhancements. I agree to the other three changes but struggle to wrap my head around this one.

Reply:
At the moment, we have an automatic configuration mode, but it's implicit, which is why the default value is `auto`. `auto` means `utilizes raw_config if defined; otherwise, it uses traditional configuration mixed with raw_blocks`.

I would also like the option to use only `raw_blocks` as a more readable and flexible version of `raw_config`. `raw_blocks` enhance operations through bosh-ops and provide clearer diffs.

Currently, this can be achieved by inserting an empty configuration into `raw_config`, but this is not a good or clear solution.
It seems to me that clarifying the configuration mode is more understandable, and certainly, the default value `auto` is backward compatible.

Reply:
Okay, I can live with that. However, I think the current version of the spec is not fully correct. 
> 'auto' - utilizes raw_config if defined; otherwise, it uses traditional configuration mixed with raw_blocks
Should probably rather be
> 'auto' - utilizes raw_config if defined and mixes it with `raw_blocks`; otherwise, it uses traditional configuration mixed with `raw_blocks`

WDYT?

Reply:
Now the template is as follows:
```
if properties.ha_proxy.config_mode == "auto"
  if properties.ha_proxy.raw_config
    # ha_proxy.raw_config
  else
    # traditional/static config
  end
end

if properties.ha_proxy.raw_blocks && !properties.ha_proxy.raw_blocks.empty? -%>
  # raw blocks
end
```
Therefore, if the `config_mode` is `auto`, it will be either `raw_config` (if not empty) or static mixed with `raw_blocks`.
If the `config_mode` is other than auto (currently this can only be raw_blocks_only), then the configuration will only contain `raw_blocks`, provided they are defined.

You are right. If the `config_mode` is auto, and both `raw_config` is not empty and `raw_blocks` are defined, there could be a problem...
So, the template should be:
```
if properties.ha_proxy.config_mode == "auto"
  if properties.ha_proxy.raw_config
    # ha_proxy.raw_config
  else
    # traditional/static config
  end
end

if !properties.ha_proxy.raw_config && properties.ha_proxy.raw_blocks && !properties.ha_proxy.raw_blocks.empty?
  # raw blocks
end
```

The `config_mode: raw_blocks_only` is not used directly, but it opens the code for future development.